src/context/TodosProvider.tsx

```diff
@@ -0,0 +1,132 @@
+import React, {
+  createContext,
+  useContext,
+  ReactNode,
+  useState,
+  useMemo,
+  useEffect,
+  useCallback,
+} from 'react';
+
+import { Todo } from '../types/Todo';
+import { getTodos } from '../api/todos';
+
+type TodoContextType = {
+  todos: Todo[];
+  filteredTodos: Todo[];
+  filterType: string;
+  messageError: string;
+  query: string
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setFilteredTodos: React.Dispatch<React.SetStateAction<Todo[]>>;
+  setFilterType: React.Dispatch<React.SetStateAction<string>>;
+  setMessageError: React.Dispatch<React.SetStateAction<string>>;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  handleSubmitSent: (event: React.FormEvent<HTMLFormElement>) => void;
+};
+
+const TodoContext = createContext<TodoContextType | undefined>(undefined);
+
+const USER_ID = 12060;
+
+const ERROR_MESSAGES = [
+  'Unable to load todos',
+  'Title should not be empty',
+  'Unable to add a todo',
+  'Unable to delete a todo',
+  'Unable to update a todo',
+];
+
+enum SortType {
+  all = 'All',
+  active = 'Active',
+  completed = 'Completed',
+}
+
+export const TodoProvider: React.FC<{ children: ReactNode }> = (
+  { children },
+) => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [filteredTodos, setFilteredTodos] = useState<Todo[]>([]);
+  const [filterType, setFilterType] = useState<string>('All');
+  const [messageError, setMessageError] = useState<string>('');
+  const [query, setQuery] = useState<string>('');
+
+  useEffect(() => {
+    getTodos(USER_ID)
+      .then(allTodos => {
+        setTodos(allTodos);
+      })
+      .catch(() => {
+        setMessageError(ERROR_MESSAGES[0]);
+      });
+  }, [setTodos, setMessageError]);
+
+  useEffect(() => {
+    switch (filterType) {
+      case SortType.all:
+        setFilteredTodos(todos);
+        break;
+      case SortType.active:
+        setFilteredTodos(todos.filter(todo => !todo.completed));
+        break;
+      case SortType.completed:
+        setFilteredTodos(todos.filter(todo => todo.completed));
+        break;
+      default:
+        break;
+    }
+  }, [filterType, todos, setFilteredTodos]);
```

Comment:
this should be `useMemo` not `useEffect`, because you are making calculations based on another state to get a value.